daemon/src/db.rs

```diff
@@ -246,159 +232,205 @@ pub async fn load_cfd_by_order_id(
     order_id: OrderId,
     conn: &mut PoolConnection<Sqlite>,
 ) -> Result<Cfd> {
-    let order_uuid = serde_json::to_string(&order_id)?;
-
     let row = sqlx::query!(
         r#"
-        select
-            orders.uuid as order_id,
-            orders.initial_price as price,
-            orders.min_quantity as min_quantity,
-            orders.max_quantity as max_quantity,
-            orders.leverage as leverage,
-            orders.trading_pair as trading_pair,
-            orders.position as position,
-            orders.origin as origin,
-            orders.liquidation_price as liquidation_price,
-            orders.creation_timestamp as creation_timestamp,
-            orders.term as term,
-            orders.oracle_event_id,
-            cfds.quantity_usd as quantity_usd,
-            cfd_states.state as state
-        from cfds as cfds
-        inner join orders as orders on cfds.order_id = orders.id
-        inner join cfd_states as cfd_states on cfd_states.cfd_id = cfds.id
-        where cfd_states.state in (
+        with ord as (
             select
-              state
-              from cfd_states
-            where cfd_id = cfds.id
-            order by id desc
-            limit 1
+                id as order_id,
+                uuid,
+                trading_pair,
+                position,
+                initial_price,
+                min_quantity,
+                max_quantity,
+                leverage,
+                liquidation_price,
+                creation_timestamp_seconds as ts_secs,
+                creation_timestamp_nanoseconds as ts_nanos,
+                term_seconds as term_secs,
+                term_nanoseconds as term_nanos,
+                origin,
+                oracle_event_id
+            from orders
+        ),
+
+        cfd as (
+            select
+                ord.order_id,
+                id as cfd_id,
+                quantity_usd
+            from cfds
+                inner join ord on ord.order_id = id
+        ),
+
+        state as (
+            select
+                id as state_id,
+                cfd.order_id,
+                cfd.quantity_usd,
+                state
+            from cfd_states
+                inner join cfd on cfd.cfd_id = cfd_states.cfd_id
+            where id in (
```

Comment:
❓ Why is this `where id in` and not `where id = `? 
`max` should always return exactly one, can sqlite not handle that?